libs/wingc/src/type_check/symbol_env.rs

```diff
@@ -180,6 +237,12 @@ impl SymbolEnv {
 				}
 			}
 			let ns = if let Some(ns) = t.as_namespace() {
+				if ns.hidden && !ignore_hidden {
+					return Err(TypeError {
+						message: format!("Symbol \"{}\" is not a namespace", symb.name),
```

Comment:
Maybe we could add a unit test covering this edge case

Reply:
I invested a bit in making a test for this, solved some bugs and cleanups on the way. Thanks for this comment!